### PR TITLE
Include tip in fee

### PR DESCRIPTION
**Pull request by MitchTurner**
Closes: https://github.com/FuelLabs/fuel-vm/issues/664

### CODE REVIEW

**Comment by MitchTurner** [comment]
I feel like I'm probably forgetting something.

File: fuel-tx/src/transaction/fee.rs [hunk 1]
```diff
@@ -150,12 +161,14 @@ pub trait Chargeable: field::Inputs + field::Witnesses + field::Policies {
...
+        tip: Word,
     ) -> u128 {
-        gas_to_fee(
+        let gas_fee = gas_to_fee(
             self.min_gas(gas_costs, fee),
             gas_price,
             fee.gas_price_factor,
-        )
+        );
+        gas_fee + tip as u128
```
Comment (xgreenx) [inline fuel-tx/src/transaction/fee.rs:171]:
```suggestion
        gas_fee.saturated_add(tip as u128)
```

File: fuel-tx/src/transaction/fee.rs [hunk 2]
```diff
@@ -166,12 +179,14 @@ pub trait Chargeable: field::Inputs + field::Witnesses + field::Policies {
...
+        tip: Word,
     ) -> u128 {
-        gas_to_fee(
+        let gas_fee = gas_to_fee(
             self.max_gas(gas_costs, fee),
             gas_price,
             fee.gas_price_factor,
-        )
+        );
+        gas_fee + tip as u128
```
Comment (xgreenx) [inline fuel-tx/src/transaction/fee.rs:189]:
```suggestion
        gas_fee.saturated_add(tip as u128)
```

File: fuel-tx/src/transaction/fee.rs [hunk 3]
```diff
@@ -166,12 +179,14 @@ pub trait Chargeable: field::Inputs + field::Witnesses + field::Policies {
         gas_costs: &GasCosts,
         fee: &FeeParameters,
         gas_price: Word,
+        tip: Word,
```
Comment (xgreenx) [inline fuel-tx/src/transaction/fee.rs:182]: Maybe you want to get it from the policies

File: fuel-tx/src/transaction/fee.rs [hunk 4]
```diff
@@ -150,12 +161,14 @@ pub trait Chargeable: field::Inputs + field::Witnesses + field::Policies {
...
```
Comment (xgreenx) [inline fuel-tx/src/transaction/fee.rs:164]: Maybe you want to get it from the policies

File: fuel-tx/src/transaction/fee.rs [hunk 5]
```diff
@@ -192,8 +207,9 @@ pub trait Chargeable: field::Inputs + field::Witnesses + field::Policies {
         let total_used_gas = min_gas.saturating_add(used_gas);
         let used_fee = gas_to_fee(total_used_gas, gas_price, fee.gas_price_factor);
```
Comment (xgreenx) [inline fuel-tx/src/transaction/fee.rs:208]:
```suggestion
        let used_fee = gas_to_fee(total_used_gas, gas_price, fee.gas_price_factor).saturating_add(tip);
```

**Review by xgreenx — COMMENTED** [review]
Could you also update the `refund_when_used_gas_is_zero` test please?=)

**Review by xgreenx — APPROVED** [review]

**Pull request merged**